hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/ha/OMRetryPolicy.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.ozone.om.ha;
+
+import com.google.protobuf.ServiceException;
+import java.io.IOException;
+import java.util.HashMap;
+import org.apache.hadoop.ipc.RemoteException;
+import org.apache.hadoop.ozone.om.exceptions.OMLeaderNotReadyException;
+import org.apache.hadoop.ozone.om.exceptions.OMNotLeaderException;
+import org.apache.ratis.protocol.exceptions.ReconfigurationInProgressException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Retry Policies for OM proxies.
+ */
+public class OMRetryPolicy {
+
+  public static final Logger LOG = LoggerFactory.getLogger(
+      OMRetryPolicy.class);
+
+  private final int maxFailovers;
+
+  private final HashMap<ExceptionType, RetryAction> defaultRetryRulesMap;
+  private HashMap<ExceptionType, RetryAction> retryRulesMapByType;
+  private HashMap<Class, RetryAction> retryRulesMapByClass;
```

Comment:
I think we could improve it in the following ways:

 * defaults do not need to be instance-specific
 * variables can be declared using `Map` interface
 * specify type parameter for `Class`
 * variables can be final

```suggestion
  private static final Map<ExceptionType, RetryAction> DEFAULT_RETRY_RULES;
  private final Map<ExceptionType, RetryAction> retryRulesByType;
  private final Map<Class<? extends Exception>, RetryAction> retryRulesByClass;
```

(Needs corresponding changes where these are used.)